### PR TITLE
vim-patch:partial 6aa57295cfbe

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/6aa57295cfbe8f21c15f0671e45fd53cf990d404

omit doc/popup.txt
omit plugin/manpager.vim

partial skip runtime/doc/eval.txt (needs 8.2.{0258,0924,1544,2324,2468,2606})

skip ftplugin/julia.vim, indent/julia.vim, syntax/julia.vim (already ported in https://github.com/neovim/neovim/commit/65f32f0f195fbf7df2478f31cab345d00a6673a4)
skip syntax/scala.vim (already ported in https://github.com/neovim/neovim/commit/a92e83ac14a0a674bc5b4b1d06d6b6c9d0d20a10)

### CODE REVIEW

File: runtime/doc/eval.txt [hunk 1]
```diff
@@ -1094,7 +1093,8 @@ Floating point numbers can be written in two forms:
 	[-+]{N}.{M}[eE][-+]{exp}
 
 {N} and {M} are numbers.  Both {N} and {M} must be present and can only
-contain digits.
+contain digits, except that in |Vim9| script in {N} single quotes between
+digits are ignored.
```
Comment (seandewar) [inline runtime/doc/eval.txt:1097]:
```suggestion
contain digits.
```

File: runtime/doc/eval.txt [hunk 2]
```diff
@@ -8631,13 +8653,16 @@ stdpath({what})					*stdpath()* *E6100*
...
+str2float({string} [, {quoted}])				*str2float()*
+ 		Convert String {string} to a Float.  This mostly works the
+ 		same as when using a floating point number in an expression,
+ 		see |floating-point-format|.  But it's a bit more permissive.
+ 		E.g., "1e40" is accepted, while in an expression you need to
+ 		write "1.0e40".  The hexadecimal form "0x123" is also
+ 		accepted, but not others, like binary or octal.
+ 		When {quoted} is present and non-zero then embedded single
+ 		quotes before the dot are ignored, thus "1'000.0" is a
+ 		thousand.
```
Comment (seandewar) [inline runtime/doc/eval.txt:8664]: Needs #14611 (which might be close to merging anyway?)

Comment (seandewar) [reply runtime/doc/eval.txt:8664]: Wait, I'm confusing it with `str2nr` -- this needs some other patch :P

Comment (clason) [reply runtime/doc/eval.txt:8664]: Yes, that's why I aspirationally included it (or rather didn't mess with the successful git merge).

File: runtime/doc/eval.txt [hunk 3]
```diff
@@ -3308,10 +3309,12 @@ confirm({msg} [, {choices} [, {default} [, {type}]]])
 <		For the console, the first letter of each choice is used as
 		the default shortcut key.  Case is ignored.
 
-		The optional {default} argument is the number of the choice
-		that is made if the user hits <CR>.  Use 1 to make the first
-		choice the default one.  Use 0 to not set a default.  If
-		{default} is omitted, 1 is used.
+		The optional {type} String argument gives the type of dialog.
+		This is only used for the icon of the GTK, Mac, Motif and
```
Comment (justinmk) [inline runtime/doc/eval.txt:3313]: delete this sentence

**Review by justinmk — APPROVED** [review]

**Pull request merged**